### PR TITLE
add button to swap div

### PR DESCRIPTION
**Pull request by gtossou**

### CODE REVIEW

File: surveyapp/surveys/views.py [hunk 1]
```diff
@@ -65,6 +72,14 @@ def renderSurveyView(request, survey_link):
         context
     )
 
+
+def get_div_content(request):
+    selected_option = request.POST.get("selected_option")
+    print(selected_option)
+    context = {"selected_option": selected_option}
+    return render(request, "new_div.html", context)
```
Comment (bbelderbos) [inline surveyapp/surveys/views.py:80]: Does not work yet:

image

Looking at other render functions, it seems you need to prepend the template name with `surveys/`

File: surveyapp/surveys/views.py [hunk 2]
```diff
@@ -27,15 +27,17 @@ def createSurveyView(request):
             question.save()
     else:
         survey_form = SurveyForm()
-        question_form = QuestionForm()
+        get_div_content(request)
```
Comment (bbelderbos) [inline surveyapp/surveys/views.py:30]: Is this related to the htmx changes? If so I would expect it to be it's own endpoint?

File: surveyapp/surveys/templates/surveys/render_survey.html [hunk 3]
```diff
@@ -9,11 +9,12 @@ <h1>List of survey</h1>
     {% if options %}
       <select id="options" name="options">
         {% for key, value in options.items %}
-        <option value="{{ key }}"> {{ value }}</option>
+        <option value="{{ key }}" name="dpd_choice"> {{ value }} </option>
```
Comment (bbelderbos) [inline surveyapp/surveys/templates/surveys/render_survey.html:12]: I think you would only set the name attribute on the outer select

File: surveyapp/surveys/templates/surveys/create_survey.html [hunk 4]
```diff
@@ -1,9 +1,19 @@
...
-    <input type="submit" value="Submit">
+    <!-- {{ question_form.as_p }} -->
+    <!-- Existing form content goes here -->
+  <select hx-post="/get_div_content/" hx-trigger="change" id="select-input">
+    <option value="">Sélectionnez une option</option>
+    <option value="option1">Option 1</option>
+    <option value="option2">Option 2</option>
+    <option value="option3">Option 3</option>
+  </select>
+  <button hx-post="/get_div_content/" hx-trigger="click" id="add-div-button">Ajouter un div</button>
```
Comment (bbelderbos) [inline surveyapp/surveys/templates/surveys/create_survey.html:16]: I don't see this endpoint in urls yet?

**Review by bbelderbos — CHANGES_REQUESTED** [review]
Good progress, few things to check please, let me know if you need help ...

File: surveyapp/surveys/templates/surveys/survey_list.html [hunk 5]
```diff
@@ -1,13 +1,13 @@
 <h1>List of survey</h1>
 
- <form action="" method="post" novalidate>
+ <form action="" method="POST" novalidate>
```
[the remark this answers is not in the document]
Comment (gtossou) [reply surveyapp/surveys/templates/surveys/survey_list.html:3]: will delete
